pkg/operator/options/options_defaulting.go

```diff
@@ -0,0 +1,113 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+import (
+	"context"
+	"fmt"
+	"hash/fnv"
+	"math/rand"
+	"net/url"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork"
+	"github.com/Azure/karpenter-provider-azure/pkg/auth"
+	"github.com/Azure/karpenter-provider-azure/pkg/utils"
+
+	armopts "github.com/Azure/karpenter-provider-azure/pkg/utils/opts"
+)
+
+// Default sets the default values for the options, but only for those too complicated to be in env default (e.g., depends on other envs)
+func (o *Options) Default(ctx context.Context) error {
+	var err error
+
+	var authManager *auth.AuthManager
```

Comment:
This is a flaw in the implementation, as `authManager` is created in both here and `operator`. Given core, there is nothing much we can do(?) if we want to follow this pattern.